### PR TITLE
Gates 17-48

### PR DESCRIPTION
**Pull request by petercombs**

### CODE REVIEW

File: eldritch/test_gates.py [hunk 1]
```diff
@@ -647,5 +647,252 @@ def testOther29(self):
...
+    self.assertEqual(self.char.sanity, 2)
+    self.assertEqual(self.char.clues, 2)
+
+  def testThreeSuccesses(self):
+    side_effect = [5, 5, 5]
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(side_effect=side_effect)):
+      self.resolve_until_done()
+    self.assertEqual(self.char.sanity, 3)
+    self.assertEqual(self.char.clues, 3)
+
```
Comment (fingoldfish) [inline eldritch/test_gates.py:895]: Can you also add a test for Other48?

**Review by fingoldfish — CHANGES_REQUESTED** [review]

File: eldritch/gate_encounters.py [hunk 2]
```diff
@@ -379,6 +620,363 @@ def Other29(char) -> events.Event:
...
+def Pluto48(char) -> events.Event:
+  return events.Nothing()
+
+
+def Other48(char) -> events.Event:
+  knowledge = events.Sequence([
+      events.Draw(char, "spells", 1),
+      events.Loss(char, {"sanity": 1})
+  ])
+  return events.PassFail(char, events.Check(char, "lore", -2), events.Nothing(), knowledge)
```
[the remark this answers is not in the document]
Comment (petercombs) [reply eldritch/gate_encounters.py]: Somehow being flooded with knowledge must have felt worse to me than "Nothing happens".

File: eldritch/gate_encounters.py [hunk 3]
```diff
@@ -363,6 +363,260 @@ def Other16(char) -> events.Event:
...
+      values.Calculation(char, "stamina"),
+      events.Gain(char, {"clues": values.Calculation(dice, "successes")}),
+      events.Nothing()
+  )
+  return events.Sequence([dice, loss, final], char)
+
+
+def Plateau28(char) -> events.Event:
+  check = events.Check(char, "sneak", -1)
+  rites = events.Loss(char, {"sanity": 3, "stamina": 3})
```
Comment (fingoldfish) [inline eldritch/gate_encounters.py:600]: If you fail the check, you are also delayed.

File: eldritch/gate_encounters.py [hunk 4]
```diff
@@ -379,6 +633,362 @@ def Other29(char) -> events.Event:
...
+      char, events.Check(char, "fight", -1), events.Nothing(), events.Delayed(char)
+  )
+
+
+def SunkenCity34(char) -> events.Event:
+  check = events.Check(char, "fight", -2)
+  shadow = events.Sequence([
+      events.Loss(char, {"stamina": 2}),
+      events.Delayed(char)
+  ])
```
Comment (fingoldfish) [inline eldritch/gate_encounters.py:706]: Attach the character to this sequence

File: eldritch/gate_encounters.py [hunk 5]
```diff
@@ -379,6 +633,362 @@ def Other29(char) -> events.Event:
...
+def Other30(char) -> events.Event:
+  return events.Loss(char, {"stamina": 1})
+
+
+def SunkenCity31(char) -> events.Event:
+  check = events.Check(char, "luck", 0)
+  pit = events.Sequence([
+      events.Loss(char, {"sanity": 1}),
+      events.Delayed(char),
+  ])
```
Comment (fingoldfish) [inline eldritch/gate_encounters.py:655]: Attach the character to this sequence.

File: eldritch/gate_encounters.py [hunk 6]
```diff
@@ -379,6 +633,362 @@ def Other29(char) -> events.Event:
...
+      char, events.Check(char, "luck", -1),
+      events.Return(char, char.place.info.name), events.Delayed(char)
+  )
+
+
+def SunkenCity41(char) -> events.Event:
+  die = events.DiceRoll(char, 1)
+  seq = events.Sequence([
+      die, events.Loss(char, {"sanity": values.Calculation(die, "sum")})
+  ])
```
Comment (fingoldfish) [inline eldritch/gate_encounters.py:832]: Attach the character to this sequence.

File: eldritch/gate_encounters.py [hunk 7]
```diff
@@ -379,6 +633,362 @@ def Other29(char) -> events.Event:
...
+def Other43(char) -> events.Event:
+  return events.MonsterAppears(char)
+
+
+def SunkenCity44(char) -> events.Event:
+  check = events.Check(char, "will", 0)
+  visage = events.Sequence([
+      events.Loss(char, {"stamina": 1}),
+      events.Delayed(char)
+  ])
```
Comment (fingoldfish) [inline eldritch/gate_encounters.py:886]: Attach the character to this sequence.

File: eldritch/gate_encounters.py [hunk 8]
```diff
@@ -379,6 +633,362 @@ def Other29(char) -> events.Event:
...
+      events.Nothing(), events.LoseItems(char, 1, decks={"spells"})
+  )
+
+
+def Pluto47(char) -> events.Event:
+  check = events.Check(char, "sneak", -2)
+  space_mead = events.Sequence([
+      events.Gain(char, {"clues": 2}),
+      events.Return(char, char.place.info.name),
+  ])
```
Comment (fingoldfish) [inline eldritch/gate_encounters.py:944]: Attach the character to this sequence.

File: eldritch/gate_encounters.py [hunk 9]
```diff
@@ -379,6 +633,362 @@ def Other29(char) -> events.Event:
...
+
+def Pluto48(char) -> events.Event:
+  return events.Nothing()
+
+
+def Other48(char) -> events.Event:
+  knowledge = events.Sequence([
+      events.Draw(char, "spells", 1),
+      events.Loss(char, {"sanity": 1})
+  ])
```
Comment (fingoldfish) [inline eldritch/gate_encounters.py:988]: Attach the character to this sequence.

File: eldritch/test_gates.py [hunk 10]
```diff
@@ -647,5 +647,303 @@ def testOther29(self):
...
+      self.resolve_until_done()
+    self.assertEqual(self.char.place.name, "Dreamlands2")
+
+  def testPass2(self):
+    self.char.place = self.state.places["Dreamlands2"]
+    self.state.event_stack.append(gate_encounters.Dreamlands23(self.char))
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(return_value=5)):
+      self.resolve_until_done()
+    self.assertEqual(self.char.place.name, "Lost")
+
```
Comment (fingoldfish) [inline eldritch/test_gates.py:703]: A few more tests I would like to see: Other23, Other24, and Dreamlands28 (make sure being devoured works).

**Review by fingoldfish — CHANGES_REQUESTED** [review]

File: eldritch/gate_encounters.py [hunk 11]
```diff
@@ -363,6 +363,260 @@ def Other16(char) -> events.Event:
...
+
+
+def GreatHall17(char) -> events.Event:
+  check = events.Check(char, "luck", -2)
+  tome = events.Draw(char, "unique", 1, target_type=items.Tome)
+  return events.PassFail(char, check, tome, events.Nothing())
+
+
+def Other17(char) -> events.Event:
+  return events.Draw(char, "spell", 1)
```
Comment (fingoldfish) [inline eldritch/gate_encounters.py:379]: Should be "spells", not "spell"

**Review by fingoldfish — APPROVED** [review]

**Pull request merged**